test/unittest/reduction/reduction_test.cpp

```diff
@@ -73,7 +70,13 @@ void run_test(const combination_t<scalar_t> combi) {
   std::vector<data_t> out_v_gpu(out_size);
   std::vector<data_t> out_v_cpu(out_size);
 
-  fill_random(in_m);
+  if (op == operator_t::Product && (rows == 513 || cols == 8195)) {
```

Comment:
Is it particularly for this size or larger size? The former can be a bug. If it is the latter, the issue is related to a limited size that a floating-point operation can accommodate. So the ad-hoc if is not the solution. You can either reduce the size of the test when the type is float in all cases or for the entire reduction test control the random value via fill_random_with_Range as depending on the operation this issue can happen in all cases. So the test should be defined carefully to consider floating-point limitation when the type is float.

Reply:
Thanks @mehdi-goli . The tests fail for larger sizes too, so I the issue appears to be overflow. I changed the range for all product tests to workaround this issue and added a comment.